### PR TITLE
Initial release process documentation

### PR DESCRIPTION
**Pull request by jt-nti**
Signed-off-by: James Taylor <jamest@uk.ibm.com>

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -2,16 +2,62 @@
 
 The following artifacts are created as a result of releasing the Hyperledger Fabric Gateway SDKs:
 
-- TBC
+- npm modules
+    - [@hyperledger/fabric-gateway](https://www.npmjs.com/package/@hyperledger/fabric-gateway)
+- Java libraries
+    - [fabric-gateway](https://search.maven.org/search?q=a:fabric-gateway)
```
Comment (bestbeforetoday) [inline RELEASING.md:8]: I think link might work better: https://search.maven.org/artifact/org.hyperledger.fabric/fabric-gateway

File: RELEASING.md [hunk 2]
```diff
@@ -2,16 +2,62 @@
...
+
+The Hyperledger Fabric Gateway SDKs follow the [Go module version numbering system](https://go.dev/doc/modules/version-numbers)
+
+## Updating version numbers
+
+The following files need to be modified when updating the version number, and these are checked by the build process to ensure they match a tagged release:
+
+- The `GATEWAY_VERSION` variable in `ci/azure-pipelines.yml`
+- The `version` element in `java/pom.xml`
+- The `version` property in `node/package.json`
```
Comment (bestbeforetoday) [inline RELEASING.md:55]: If the samples are staying in this repo then, for Node and Java, probably we should update both their versions and fabric-gateway dependency versions

Comment (jt-nti) [reply RELEASING.md:55]: > Update test, sample, and docs files to match the new version

I was admittedly/deliberately a bit vague about the sample versions but those sort of things tend to change over time, whereas the versions here are a bit more fundamental

If the samples are staying (are they?), I'm not sure we should be updating the version of the samples (that doesn't happen in fabric-samples afaik), and I think it's only the Java dependency that needs updating and it turns out that the build breaks if you miss it :)

Comment (bestbeforetoday) [reply RELEASING.md:55]: I was thinking the sample version matching up to a fabric-gateway version would help to identify issues with mismatched version not working due to API changes. But I guess you're right, the samples are checked in with the code they match and you shouldn't really be using them as-is outside of that context. And I'm not convinced keeping the samples here is the best idea either.

**Review by bestbeforetoday — APPROVED** [review]

**Pull request merged**